### PR TITLE
Fix structure analysis to make it work on unit test with "while" loop with break at the end of loop body

### PR DESCRIPTION
**Pull request by ptomin**
- Create structure analysis unit test with "while" loop with break at the end of loop body
- Virtualize loop break edges in different ways for "while" and "do-while"

### CODE REVIEW

File: subjects/PDP-11/lunar.c [hunk 1]
```diff
@@ -174,23 +174,22 @@ void fn0444()
...
-		do
+		while (65356 - r0_15 <= 0x00)
 		{
-			if (65356 - r0_15 > 0x00)
-				goto l0464;
 			r0_15 = r0_15 + 0x0168;
-		} while (r0_15 > 0x00);
+			if (r0_15 > 0x00)
+				break;
+		}
```
Comment (ptomin) [inline subjects/PDP-11/lunar.c]: Condition should be inverted. But it's another issue

**Comment by ptomin** [comment]
Merged. Are you agree, John?

**Comment by uxmal** [comment]
Мне нравится.  I do want to look closer at the pdp-11 regression but I will be away from my computer until the evening of 1 Jan .

**Comment by ptomin** [comment]
> I do want to look closer at the pdp-11 regression

Do you mean  https://github.com/uxmal/reko/pull/525#discussion_r159122198? I have fixed it on my local machine.

**Comment by uxmal** [comment]
Yes. There are some other issues there as well that are not related to structure reconstruction.

**Comment by ptomin** [comment]
I hope pdp-11 regression was fixed in 526

**Pull request merged**